### PR TITLE
feat(#824): Add `JNA` Integration Test

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
In this PR I successfully added an integration test that uses `JNA` library. This PR proves that 
`jeo-maven-plugin` works well with `JNA` classes.

Closes: #824.

---

## PR-Codex overview
This PR focuses on enhancing the `JNA` integration test by improving the bytecode representation and instruction handling, renaming methods for clarity, and adding documentation. It also includes new test cases for better coverage and updates to the `pom.xml` for project configuration.

### Detailed summary
- Added `view()` method to several bytecode classes for better human-readable output.
- Renamed `isGoto()` to `isJump()` in multiple classes to improve clarity.
- Introduced a new method `instructionsView()` in `BytecodeMethod` for displaying instructions.
- Enhanced `BytecodeMethodTest` with parameterized tests for real bytecode methods.
- Updated `invoker.properties` and `pom.xml` for improved integration testing configuration.
- Added a `README.md` for `JNA` usage with instructions for running tests.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you take a look please? This test tries to resemble the issue you have here https://github.com/objectionary/hone-maven-plugin/issues/58
As you can see, everything works just fine. Please, correct me, if the test I wrote is different from what you have in `OptimizeMojoTest#optimizesJnaClasses`, in this case I will try to reproduce the same scenario.

As for the https://github.com/objectionary/hone-maven-plugin/issues/58 issue, it seems, that when you start testing `mvn test` the `JNA` library is just missing in the classpath.

**Comment by yegor256** [comment]
@volodya-lombrozo this way it works, but try to replace the `<dependency/>` in the `pom.xml` with this:

```
      <plugin>
        <artifactId>maven-dependency-plugin</artifactId>
        <version>3.8.1</version>
        <executions>
          <execution>
            <phase>generate-resources</phase>
            <goals>
              <goal>unpack</goal>
            </goals>
            <configuration>
              <artifactItems>
                <artifactItem>
                  <groupId>net.java.dev.jna</groupId>
                  <artifactId>jna</artifactId>
                  <version>5.15.0</version>
                  <type>jar</type>
                </artifactItem>
              </artifactItems>
              <outputDirectory>${project.build.outputDirectory}</outputDirectory>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

Basically, you will unpack the dependency and let JEO process the `.class` files from JNA.

**Comment by yegor256** [comment]
@volodya-lombrozo use the latest version of EO: 0.40.3, it should be much faster

**Comment by volodya-lombrozo** [comment]
@yegor256 It is with the `0.40.3` version already.

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you take a look one more time, please? I fixed the original problem in this PR: https://github.com/objectionary/jeo-maven-plugin/pull/827
Here I just prove that the solution solves the original problem with JNA classes. 

Pay attention, I excluded `phi/unphi` and `unroll` from the `jna` integration test because it takes too much time. Just to compare, the 'short' path (without `phi/unphi`) takes 13 seconds to run, while the `full` path takes 31 minutes. You can run the `full` path locally, to be sure, that everything works well. I've already done it myself:

```
[INFO] Building: jna/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           jna/pom.xml ...................................... SUCCESS (1860 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 1, Failed: 0, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  31:13 min
[INFO] Finished at: 2024-10-31T12:16:46+03:00
[INFO] ------------------------------------------------------------------------
```

File: src/it/jna/pom.xml [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <stack-size>256M</stack-size>
+    <jeo.disassemble>${project.build.directory}/generated-sources/jeo-disassemble</jeo.disassemble>
+    <eo.phi>${project.build.directory}/generated-sources/eo-phi</eo.phi>
+    <eo.unphi>${project.build.directory}/generated-sources/eo-unphi</eo.unphi>
+    <jeo.unroll>${project.build.directory}/generated-sources/jeo-unroll</jeo.unroll>
+    <jeo.assemble>target/classes</jeo.assemble>
+  </properties>
+  <dependencies>
+    <dependency>
```
Comment (yegor256) [inline src/it/jna/pom.xml:47]: @volodya-lombrozo you can remove it from the list of deps, since you unpack it

**Review by yegor256 — CHANGES_REQUESTED** [review]
@volodya-lombrozo looks much better now, see one comment above

**Comment by volodya-lombrozo** [comment]
@yegor256  Indeed! You are right, thank you. Removed this library. Could you have a look one more time, please?

Also I should warn you. If you remove `target/classes` folder somewhere, it might be a cause of serious issues. For example, `jna` library keeps native `.so` files inside `target/classes` and if you remove this folder entirely - it will crash your application.  By this reason, in my test I remove only `*.class` files from `target/classes` folder:

```xml
<plugin>
  <artifactId>maven-clean-plugin</artifactId>
  <version>3.4.0</version>
  <executions>
    <execution>
      <id>clean-target-classes</id>
      <phase>process-classes</phase>
      <goals>
        <goal>clean</goal>
      </goals>
      <configuration>
        <excludeDefaultDirectories>true</excludeDefaultDirectories>
        <filesets>
          <fileset>
            <directory>target/classes</directory>
            <includes>
              <include>**/*.class</include>
            </includes>
          </fileset>
        </filesets>
      </configuration>
    </execution>
  </executions>
</plugin>
```

**Review by yegor256 — APPROVED** [review]
@volodya-lombrozo LGTM

**Pull request merged**